### ISSUE TITLE
ERROR - Unable to set I2P Socket options to non-blocking, after I2P accept was issued.

### ISSUE THREAD

**Issue by entertheabyss** [Anoncoin/oldrepo-backup#105]
i have been having issues with my client
1) when stopping the client it does not stop and continues to run until i kill it manually
2) i noticed that the client will stop downloading new blocks from peers and gets stuck even though getpeerinfo shows several connections

im not sure if i should make this 2 seperate issues
but it might be caused when i2prouter is restarted while anoncoin is running?

**Comment by Cryptoslave** [comment]
Yes, i2prouter should not be restarted when anoncoin is running.

We had the issue last year with this exact cause

**Comment by entertheabyss** [comment]
hmm, is there any way to get anoncoin to retry to connect.
or at least throw up an error in getinfo as opposed to just in the debug log?
